### PR TITLE
#6449 Introduce EventConfig

### LINKED ISSUE #6449: Introduce `EventConfig`
### `EventConfig`
* `maxEventQueueForCons`
* `eventIntakeQueueThrottleSize`
* `eventIntakeQueueSize`
* `randomEventProbability`
* `staleEventPreventionThreshold`
* `rescueChildlessInverseProbability`
* `eventStreamQueueCapacity`
* `eventsLogPeriod`
* `eventsLogDir`
* `enableEventStreaming`

- [ ] Introduce EventConfig
- [ ] Add Mappings for the new names

### PR DESCRIPTION
**Pull request by timo0**
Fix #6449

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/SyncManagerTest.java [hunk 1]
```diff
@@ -110,10 +112,11 @@ public EventCreationRuleResponse shouldCreateEvent(BaseEvent selfParent, BaseEve
                     return null;
                 }
             };
-            final ReconnectConfig config = new TestConfigBuilder()
+            Configuration configuration = new TestConfigBuilder()
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/SyncManagerTest.java:115]:
```suggestion
            final Configuration configuration = new TestConfigBuilder()
```

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/eventhandling/ConsensusRoundHandlerTests.java [hunk 2]
```diff
@@ -186,12 +188,17 @@ private void initConsensusHandler(final SwirldState swirldState) {
         final State state = new State();
         state.setSwirldState(swirldState);
 
-        when(settingsProvider.getMaxEventQueueForCons()).thenReturn(500);
-
         final AddressBook addressBook = new RandomAddressBookGenerator().build();
 
+        Configuration configuration = new TestConfigBuilder()
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/eventhandling/ConsensusRoundHandlerTests.java:193]:
```suggestion
        final Configuration configuration = new TestConfigBuilder()
```

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/eventhandling/ConsensusRoundHandlerTests.java [hunk 3]
```diff
@@ -186,12 +188,17 @@ private void initConsensusHandler(final SwirldState swirldState) {
...
+                .withValue("event.maxEventQueueForCons", 500)
+                .getOrCreateConfig();
+        PlatformContext platformContext = TestPlatformContextBuilder.create()
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/eventhandling/ConsensusRoundHandlerTests.java:196]:
```suggestion
        final PlatformContext platformContext = TestPlatformContextBuilder.create()
```

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/components/EventTaskCreatorTest.java [hunk 4]
```diff
@@ -84,16 +86,31 @@ void newMocks() {
                 selfId,
                 eventIntakeMetrics,
                 eventQueueThread,
-                setting,
+                config,
                 syncManager,
                 () -> random);
     }
 
+    private EventConfig configRandomEventProbability() {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/components/EventTaskCreatorTest.java:94]: nonnull Annotation for result

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/components/EventTaskCreatorTest.java [hunk 5]
```diff
@@ -84,16 +86,31 @@ void newMocks() {
...
+        return new TestConfigBuilder()
+                .withValue("event.randomEventProbability", 1)
+                .getOrCreateConfig()
+                .getConfigData(EventConfig.class);
+    }
+
+    private EventConfig configRescueChildlessInverseProbability() {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/components/EventTaskCreatorTest.java:101]: nonnull Annotation for result

**Review by cody-littley — DISMISSED** [review]
LGTM, but we should probably prioritize fixing the `Settings.java` log warnings before we remove more settings from that location.

**Comment by timo0** [comment]
@cody-littley please see  https://github.com/hashgraph/hedera-services/pull/7075

**Review by cody-littley — APPROVED** [review]

**Review by hendrikebbers — DISMISSED** [review]

**Review by cody-littley — DISMISSED** [review]

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/components/EventTaskCreatorTest.java [hunk 6]
```diff
@@ -101,9 +105,9 @@ private EventConfig configRandomEventProbability() {
     }
 
     @NonNull
-    private EventConfig configRescueChildlessInverseProbability() {
+    private EventConfig configRescueChildlessInverseProbability(int value) {
```
Comment (cody-littley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/components/EventTaskCreatorTest.java:108]:
```suggestion
    private EventConfig configRescueChildlessInverseProbability(final int value) {
```

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/components/EventTaskCreatorTest.java [hunk 7]
```diff
@@ -166,11 +165,21 @@ void testEventRescue() throws InterruptedException {
...
 
-        reset(eventQueueThread);
+    @Test
+    @Tag(TestTypeTags.FUNCTIONAL)
+    @Tag(TestComponentTags.PLATFORM)
+    @DisplayName("test addEvent()")
+    void testEventNotRescue() throws InterruptedException {
+        init(configRescueChildlessInverseProbability(0));
+
+        EventImpl eventToRescue = mock(EventImpl.class);
```
Comment (cody-littley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/components/EventTaskCreatorTest.java:177]:
```suggestion
        final EventImpl eventToRescue = mock(EventImpl.class);
```

**Review by hendrikebbers — APPROVED** [review]

**Review by cody-littley — APPROVED** [review]

**Pull request merged**